src/core/unit.c

```diff
@@ -6397,22 +6397,21 @@ Condition *unit_find_failed_condition(Unit *u) {
         return failed_trigger && !has_succeeded_trigger ? failed_trigger : NULL;
 }
 
-int unit_can_live_mount(const Unit *u, sd_bus_error *error) {
+int unit_can_live_mount(Unit *u, sd_bus_error *error) {
         assert(u);
 
         if (!UNIT_VTABLE(u)->live_mount)
                 return sd_bus_error_setf(
                                 error,
-                                SD_BUS_ERROR_INVALID_ARGS,
-                                "Live mounting not supported for unit type '%s' of unit '%s'.",
-                                unit_type_to_string(u->type),
-                                u->id);
+                                SD_BUS_ERROR_NOT_SUPPORTED,
+                                "Live mounting not supported by unit type '%s'",
```

Comment:
Why the trailing period is dropped?

Reply:
So I think our bus errors shouldn't contain periods, since they could be embedded in arbitrary strings by clients. Our logs however should definitely contain the period.

Reply:
I didn't mean "arbitrary" arbitrary, but something like "bus method failed with: %s, terminating". Period in the middle of the sentence feels really weird to me. I've started to remove the period in bus error message I touch for a while now, and I'd prefer to add this to our coding style some day.

Reply:
In such case, something like "bus method Foo failed, terminating the process: %s" should be OK.
Or, usually, we say something like "Failed to call Foo bus method, proceeding anyway, ignoring: %s". There is no problem that bus errors end with period.

Note, strerror() and friends always provide an error message that ends with period. For consistency, all bus error messages should ends with period.

Reply:
Again, the log message is delivered to clients which aren't necessarily controled by us...

> Note, strerror() and friends always provide an error message that ends with period.

Hmm? IIRC that's not the case. Anyways, I'm tempted to just merge this... We can still change this systematically later if desirable.

Reply:
> I'm tempted to just merge this... We can still change this systematically later if desirable.

Then, please do not touch any existing log messages, especially something like insignificant but triggers potentially long debates... We can systematically drop periods at the end from bus error messages later...

Reply:
I didn't expect any of these. https://github.com/systemd/systemd/pull/33093/commits/759ac0dec33bbd9d307f19e4565d4548215fe5b4 was a previous attempt that went in smoothly.